libshortfin/CMakeLists.txt

```diff
@@ -46,9 +39,77 @@ list(APPEND CMAKE_MODULE_PATH
 include(shortfin_library)
 
 # Dependencies.
-find_package(spdlog)
-find_package(xtensor)
-find_package(IREERuntime)
+
+## fmt
+FetchContent_Declare(
+  fmt
+  GIT_REPOSITORY https://github.com/fmtlib/fmt.git
+  GIT_TAG        e69e5f977d458f2650bb346dadf2ad30c5320281 # 10.2.1 (sync with spdlog)
+  # Do not download if package is already available
+  FIND_PACKAGE_ARGS NAMES fmt
+)
+
+## spdlog
+set(SPDLOG_FMT_EXTERNAL ON)
+FetchContent_Declare(
+  spdlog
+  GIT_REPOSITORY https://github.com/gabime/spdlog.git
+  GIT_TAG        2d4acf8cc321d7783d8f2e22e17a794c6d0e9450 # v1.14.1
+  # Do not download if package is already available
+  FIND_PACKAGE_ARGS NAMES spdlog
+)
+
+## xtl: required for xtensor
+FetchContent_Declare(
+  xtl
+  GIT_REPOSITORY https://github.com/xtensor-stack/xtl.git
+  GIT_TAG        a7c1c5444dfc57f76620391af4c94785ff82c8d6 # v0.7.7
+  # Do not download if package is already available
+  FIND_PACKAGE_ARGS NAMES xtl
+)
+
+## xtensor
+FetchContent_Declare(
+  xtensor
+  GIT_REPOSITORY https://github.com/xtensor-stack/xtensor.git
+  GIT_TAG        3634f2ded19e0cf38208c8b86cea9e1d7c8e397d # v0.25.0
+  # Do not download if package is already available
+  FIND_PACKAGE_ARGS NAMES xtensor
+)
+
+FetchContent_MakeAvailable(fmt spdlog xtl xtensor)
+
+## iree runtime
+
+set(SHORTFIN_IREE_SOURCE_DIR "" CACHE FILEPATH "Path to IREE source")
+if(SHORTFIN_IREE_SOURCE_DIR)
+  set(IREE_BUILD_COMPILER OFF)
+  set(IREE_BUILD_TESTS OFF)
+  add_subdirectory(${SHORTFIN_IREE_SOURCE_DIR} iree SYSTEM)
```

Comment:
Let's also only enable backends that we care about: right now just local and amdgpu.

Also, for my own edification, why do you need SYSTEM here?

Reply:
Following the discussion on discord, seems like there is a `unused function` error propagating from IREE and due to `-Wno-used-function` set. However, just building with `IREE_BUILD_COMPILER_OFF` shouldn't trigger that error (we don't see it in iree-bare-metal-arm) but I can dig into with @Groverkss tomorrow.

Reply:
The error comes because IREE uses -Wno-unused-function and other compilation options, while our CMakeLists.txt doesn't (for now at least). IREE will fail to compile without these flags. I used SYSTEM to isolate IREE's compilation flags to itself. I'm guessing this could also bite us in future if we wanted to use some of our own flags. Not sure whats preferred here.

Reply:
Let's go with what you have (at least) for now :)